core/kernel/users/class.GenerisUser.php

```diff
@@ -75,32 +75,53 @@ public function getPropertyValues($property)
         } elseif (!isset($this->cache[$property])) {
             $this->cache[$property] = $this->getUncached($property);
         }
+
         return $this->cache[$property];
     }
 
     private function getUncached($property)
     {
-        $value = [];
         switch ($property) {
             case GenerisRdf::PROPERTY_USER_DEFLG:
+                $lang = $this->findLanguage($property);
+                return !empty($lang) ? $lang : [DEFAULT_LANG];
```

Comment:
Are you sure we want to change it? Is it not just UI language in the scope?

Reply:
I've rechecked this, and I think it is correct as it is now. Still, is a bit tricky for me after all the recent changes, so let me explain my rationale:

Previously we handled both `PROPERTY_USER_DEFLG` and `PROPERTY_USER_UILG` the same way (that is, using the logic that is now in `findLanguage()` **and also** returning `DEFAULT_LANG` if the language was not set).  We moved that logic to a common method because it applies to both cases but now we are changing the return value on the error case for one of them.

`findLanguage()` does **not** return `DEFAULT_LANG` if `$property` is not set for the user, so what we are doing now is passing the return value from `findLanguage()` directly for `PROPERTY_USER_UILG` (so `UserLanguageService::getInterfaceLanguage()` [[here](https://github.com/oat-sa/generis/blob/01fe296d5c83561999a011bc6a230708d7323085/common/oatbox/user/UserLanguageService.php#L60)] receives it as `null` if not set).
- In that case, the caller (`UserLanguageService::getInterfaceLanguage()`) will pick its return return  value from `OPTION_INTERFACE_LANGUAGE`.

Now, for this case (`PROPERTY_USER_DEFLG`), `findLanguage()` behaves the same way (returns `null` if the language is not set) **but** we need to return the `DEFAULT_LANG` from here to maintain the old behavior.
- Therefore, we keep returning `DEFAULT_LANG` explicitly here for the "error" case of `findLanguage()` (i.e. when it returns `null`).

Do you think we are missing something?

Reply:
Thanks for the explanation :) 

But aren't we addressing in this task is only the _interface language_? Why should we escalate this to the default language? Also, keep in mind that when @ktsirkou changed the default language for UDIR installation we had many problems parsing the RDF records and the UI broke.

Reply:
You are right. The return value will be null and we will apply the change. Closing the thread :)